sympy/assumptions/handlers/ntheory.py

```diff
@@ -246,3 +250,30 @@ def Basic(expr, assumptions):
                 return None
             return not _even
         return _integer
+
+
+AskCoprimeHandler = Dispatcher('AskCoprimeHandler')
```

Comment:
Multiple dispatch is probably a cleaner way of doing the handlers than the classes with staticmethods, although it still has the same logical limitations.

Reply:
Yes, I think so. Maybe I can change every existing handler to multipledispatch in this PR. How's that?

Reply:
It will probably be a lot of changes so maybe it will be better for a separate PR.

Reply:
Then I will mark this as resolved.